convert2rhel/breadcrumbs.py

```diff
@@ -152,50 +165,27 @@ def _generate_json(self):
             "run_id": self.run_id,
         }
 
-        write_obj_to_array_json(path=FILE, new_object=data, key="activities")
-
+    def _save_migration_results(self):
+        """Write the results of the breadcrumbs to the migration-results file."""
+        loggerinst.info("Writing breadcrumbs to '%s'.", MIGRATION_RESULTS_FILE)
+        _write_obj_to_array_json(path=MIGRATION_RESULTS_FILE, new_object=self.data, key="activities")
 
-def sanitize_cli_options(all_cli_options, options_to_sanitize):
-    """Change value of CLI options to asterisks to hide sensitive information.
+    def _save_rhsm_facts(self):
+        """Write the results of the breadcrumbs to the rhsm custom facts file."""
+        data = utils.flatten(dictionary=self.data, parent_key=RHSM_CUSTOM_FACTS_NAMESPACE)
+        try:
+            loggerinst.info("Writing RHSM custom facts to '%s'.", RHSM_CUSTOM_FACTS_FILE)
+            # We don't need to use `_write_obj_to_array_json` function here, because
+            # we only care about dumping the facts without having multiple copies of
+            # it.
+            utils.write_json_object_to_file(path=RHSM_CUSTOM_FACTS_FILE, data=data)
+        except (IOError, OSError):
+            rhsm_facts_path = os.path.dirname(RHSM_CUSTOM_FACTS_FILE)
+            loggerinst.warning("Unable to find RHSM facts folder at '%s'.", rhsm_facts_path)
```

Comment:
I think that if the folder /etc/rhsm/facts/ does not exist, we should create it before trying to write the facts file to it.

The way it is now, we will have the breadcrumbs file available only after we install the subscription-manager package. But we'd like to know about even about the failures before we install the subscription-manager package. As a matter of fact, the subscription-manager pkg is not even being installed when the user runs the tool with the `--norhsm` option.

The facts are supposed to be collected by sos. We should check whether the sos plugin collects them even without the subscription-manager installed.

<details>
  <summary>Example of where the breadcrumbs should have been written</summary>

```
[10/27/2022 22:39:25] TASK - [Prepare: Checking if the loaded kernel version is the most recent] 
[10/27/2022 22:39:25] DEBUG - Calling command 'repoquery --quiet --qf "%{BUILDTIME}\t%{VERSION}-%{RELEASE}\t%{REPOID}" kernel'
[10/27/2022 22:39:25] DEBUG - Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/convert2rhel/main.py", line 94, in main
    checks.perform_pre_checks()
  File "/usr/lib/python2.6/site-packages/convert2rhel/checks.py", line 68, in perform_pre_checks
    is_loaded_kernel_latest()
  File "/usr/lib/python2.6/site-packages/convert2rhel/checks.py", line 672, in is_loaded_kernel_latest
    _, latest_kernel, repoid = packages[0]
ValueError: need more than 1 value to unpack

Writing breadcrumbs to '/etc/migration-results'.
[10/27/2022 22:39:25] DEBUG - Prior migration log: {"activities": [{"executed": "/usr/bin/convert2rhel -y --debug --no-rpm-va --serverurl subscription.rhsm.stage.redhat.com --username c2r_main --password ***** --pool 2c94a05683f64f030184162087bf18cb", "success": false, "run_id": "null", "activity_ended": "2022-10-27T20:36:11.097896Z", "activity_started": "2022-10-27T20:36:01.609912Z", "version": "1", "env": {}, "activity": "conversion", "source_os": {"version": "6.10", "id": "Final", "name": "CentOS"}, "packages": [{"nevra": "convert2rhel-1.0-1.20221027201547923673.pr587.61.gda871a6.el6.noarch", "signature": "RSA/8, Thu Oct 27 22:17:20 2022, Key ID 175e6797ecf2dde9"}], "target_os": "null"}]}
[10/27/2022 22:39:25] DEBUG - Updated migration log: {"activities": [{"executed": "/usr/bin/convert2rhel -y --debug --no-rpm-va --serverurl subscription.rhsm.stage.redhat.com --username c2r_main --password ***** --pool 2c94a05683f64f030184162087bf18cb", "success": false, "run_id": "null", "activity_ended": "2022-10-27T20:36:11.097896Z", "activity_started": "2022-10-27T20:36:01.609912Z", "version": "1", "env": {}, "activity": "conversion", "source_os": {"version": "6.10", "id": "Final", "name": "CentOS"}, "packages": [{"nevra": "convert2rhel-1.0-1.20221027201547923673.pr587.61.gda871a6.el6.noarch", "signature": "RSA/8, Thu Oct 27 22:17:20 2022, Key ID 175e6797ecf2dde9"}], "target_os": "null"}, {"executed": "/usr/bin/convert2rhel -y --debug --no-rpm-va --serverurl subscription.rhsm.stage.redhat.com --username c2r_main --password ***** --pool 2c94a05683f64f030184162087bf18cb", "run_id": "null", "packages": [{"nevra": "convert2rhel-1.0-1.20221027201547923673.pr587.61.gda871a6.el6.noarch", "signature": "RSA/8, Thu Oct 27 22:17:20 2022, Key ID 175e6797ecf2dde9"}], "target_os": "null", "success": false, "activity_ended": "2022-10-27T20:39:25.381003Z", "version": "1", "env": {}, "activity": "conversion", "source_os": {"version": "6.10", "id": "Final", "name": "CentOS"}, "activity_started": "2022-10-27T20:39:19.906445Z"}]}
Writing RHSM custom facts to '/etc/rhsm/facts/convert2rhel.facts'.
WARNING - Unable to find RHSM facts folder at '/etc/rhsm/facts'.
No changes were made to the system.
```
</details>